woocommerce/wc-admin
# Feature/7319 add woocommerce com data

Addresses #7319 

Integrate the `payment-method-promotions.json` endpoint from woocommerce.com into the payment method table for wc pay. This will make use of the title and description from woocommerce.com.

No changelog as this will be added in the final PR

### Screenshots

<img width="1454" alt="Screen Shot 2021-08-27 at 3 44 28 PM" src="https://user-images.githubusercontent.com/2240960/131174683-c3edd596-db11-4086-810c-6d91de7513d3.png">

### Detailed test instructions:

*Make sure you don't have WooCommerce Payments installed or enabled.
- Have a local version of woocommerce.com running using the branch of this PR (`add/payment-method-recommendation-endpoint`) -> 11084-gh-Automattic/woocommerce.com
- Change the url in the [`DataSourcePoller.php`](https://github.com/woocommerce/woocommerce-admin/blob/feature/7319_add_woocommerce_com_data/src/Features/WCPayPromotion/DataSourcePoller.php#L24) to point to your local version with to `http://woocommerce.test` or using your local ip might work as well - `http://192....`
you might have to remove the `woocommerce_admin_payment_method_promotion_specs` transient.
- Go to **WooCommerce > Settings > Payments** and see if the description shows up correctly.
- Update the url to something invalid and clear the transient again
- Go back to the payment settings screen
- The psuedo woocommerce payment method should not be listed in the payment method table now.
